pkg/gateway/model_build_targetgroup.go

```diff
@@ -240,36 +250,43 @@ func (t *targetGroupModelBuildTask) BuildTargetGroup(ctx context.Context) error
 			HealthCheckConfig:   healthCheckConfig,
 			IpAddressType:       ipAddressType,
 		},
-	}
-
-	tg := latticemodel.NewTargetGroup(t.stack, tgName, tgSpec)
-	t.log.Infof("buildTargetGroup, tg[%s], tgSpec %v", tgName, tg)
+	})
 
-	// add targetgroup to localcache for service reconcile to reference
-	// for serviceexport, the httproutename is set to ""
-	t.datastore.AddTargetGroup(tgName, "", "", "", tgSpec.Config.IsServiceImport, "")
+	t.log.Infow("stackTG:",
+		"targetGroupName", stackTG.Spec.Name,
+		"K8SServiceName", stackTG.Spec.Config.K8SServiceName,
+		"K8SServiceNamespace", stackTG.Spec.Config.K8SServiceNamespace,
+		"Protocol", stackTG.Spec.Config.Protocol,
+		"ProtocolVersion", stackTG.Spec.Config.ProtocolVersion,
+		"IpAddressType", stackTG.Spec.Config.IpAddressType,
+		"HealthCheckConfig", stackTG.Spec.Config.HealthCheckConfig,
+	)
+
+	t.datastore.AddTargetGroup(targetGroupName, "", "", "", false, "")
+	t.datastore.SetTargetGroupByServiceExport(targetGroupName, false, true)
+	return stackTG, nil
+}
 
-	if !t.serviceExport.DeletionTimestamp.IsZero() {
-		// triggered by serviceexport delete
-		t.datastore.SetTargetGroupByServiceExport(tgName, false, false)
-	} else {
-		// triggered by serviceexport add
-		t.datastore.SetTargetGroupByServiceExport(tgName, false, true)
+func (t *targetGroupModelBuildTask) buildTargetGroupForServiceExportDeletion(ctx context.Context, targetGroupName string) (*latticemodel.TargetGroup, error) {
+	stackTG := latticemodel.NewTargetGroup(t.stack, targetGroupName, latticemodel.TargetGroupSpec{
+		Name:      targetGroupName,
+		LatticeID: "",
+		IsDeleted: true,
+	})
+	t.datastore.SetTargetGroupByServiceExport(targetGroupName, false, false)
+	dsTG, err := t.datastore.GetTargetGroup(targetGroupName, "", false)
+	t.log.Debugf("TargetGroup cached in datastore: %v", dsTG)
+	if err != nil {
+		return nil, fmt.Errorf("%w: targetGroupName: %s", err, targetGroupName)
 	}
```

Comment:
log after error handling. if err != nil {} then log.debug.

Reply:
`return nil, fmt.Errorf()`  already shown the error message? Hope to print `dsTG` for both `err == nil` and `err != nil` for more detailed debug log

Reply:
When there is result and error from function call we dont use result if there is an error, even for logging. You never know what kind of data returned as result when there is an error, code might even panic if you try to do some null pointer de-reference. Please change order.
```go
        dsTG, err := t.datastore.GetTargetGroup(targetGroupName, "", false)
	if err != nil {
		return nil, fmt.Errorf("%w: targetGroupName: %s", err, targetGroupName)
	}
	t.log.Debugf("TargetGroup cached in datastore: %v", dsTG)
```

Reply:
Ok, good points, changed